src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -188,7 +175,9 @@ public function handle(GetOrderForViewing $query): OrderForViewing
             $this->getOrderCustomer($order),
             $this->getOrderShippingAddress($order),
             $this->getOrderInvoiceAddress($order),
-            $this->getOrderProducts($order),
+            $this->getOrderProductsForViewingHandler->handle(
```

Comment:
I think you can keep this call in a dedicated method `getOrderProducts` The content will be short but it will be cleaner